### ISSUE TITLE
no-inferred-empty-object: false negative for classes with constructor

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#195]
```ts
class MyWrap<T> {
    constructor(public value?: T) {}
}

new MyWrap(); // this one is not detected
new MyWrap<number>()
new MyWrap(1);

class MyOtherWrap<T> {
    constructor() {}
}
new MyOtherWrap(); // this one is not detected
new MyOtherWrap<number>();
```
